jetson/science_bridge/src/__main__.py

```diff
@@ -249,22 +248,44 @@ def mosfet_transmit(self, channel, msg):
         pass
 
     def nav_status(self, channel, msg):  # TODO - fix make better
-        print("Received nav req")
         # Off, Done, Else
-
         struct = NavStatus.decode(msg)
+
+        if self.previous_auton_msg == struct.nav_state_name:
+            return
+
+        print("Received new nav req")
+
         message = "$Mosfet,{device},{enable}"
 
-        # Off = Blue
+        blue_off = message.format(device=Mosfet_devices.BLUE_LED.value, enable=0)
+        blue_off = self.add_padding(blue_off)
+        self.ser.write(bytes(blue_off, encoding='utf-8'))
+        red_off = message.format(device=Mosfet_devices.RED_LED.value, enable=0)
+        red_off = self.add_padding(red_off)
+        self.ser.write(bytes(red_off, encoding='utf-8'))
+        green_off = message.format(device=Mosfet_devices.GREEN_LED.value, enable=0)
+        green_off = self.add_padding(green_off)
+        self.ser.write(bytes(green_off, encoding='utf-8'))
+
+        self.previous_auton_msg = struct.nav_state_name
+
         if struct.nav_state_name == "Off":
-            print("navstatus off")
+            prev = Auton_state.OFF
+        elif struct.nav_state_name == "Done":
+            prev = Auton_state.DONE
+        else:
+            prev = Auton_state.ELSE
+
+        # Off = Blue
+        if prev == Auton_state.OFF:
+            print("navstatus off - blue")
             blue = message.format(device=Mosfet_devices.BLUE_LED.value, enable=1)
             blue = self.add_padding(blue)
             self.ser.write(bytes(blue, encoding='utf-8'))
-            prev = Auton_state.OFF
         # Done = Flashing green
-        elif struct.nav_state_name == "Done":
-            print("navstatus Done")
+        elif prev == Auton_state.DONE:
+            print("navstatus done - green blink")
             # Flashing by turning on and off for 1 second intervals
 
             NUMBER_OF_LED_BLINKS = 6
```

Comment:
the number of blinks part is a bit strange i think, like the function gets stuck doing blinks but what if nav status is still publishing in the meantime and it goes from done to off and we try to teleop it and the array doesn't respond i time? its 12 seconds which could be significant...i would make it so the blinks are handled in the nucleo code similarly how the arduino led did it so it blinks until auton goes to off or on basically without having to constantly call the handler

Reply:
was planning on doing this yes

Reply:
sg sg